src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -579,6 +580,16 @@ public String getLoginUrl() {
         return "securityRealm/commenceLogin";
     }
 
+    @Override
+    protected String getPostLogOutUrl(StaplerRequest req, Authentication auth) {
+        // if we just redirect to the root and anonymous does not have Overall read then we will start a login all over again.
```

Comment:
So a future enhancement for the plugin author could be to make this configurable so that it sends you to github to logout there as well :)